universe/com.goal_smart/manifest.tt

```diff
@@ -75,100 +77,131 @@ class @com.goal_smart
-                          preposition=["against #"]
+                            base=["opposition team", "opposition side", "opposition group"],
+                            passive_verb=["played against #"],
+                            preposition=["against #"]
                         }],
                         out our_score: Number
                         #_[canonical = {
-                        base = ["our goals", "our points", "our score"]
+                          base = ["our goals", "our points", "our score"]
                         }],
                         out opposition_score: Number
                         #_[canonical = {
-                        base = ["opposition goals", "opposition points", "opposition score"]
+                          base = ["opposition goals", "opposition points", "opposition score"]
                         }],
                         out result: Enum(tied, won, lost)
                         #_[canonical = {
-                        base = ["result", "final", "outcome"]
+                          base = ["result", "final", "outcome"]
                         }],
                         out venue: String
                         #_[canonical = {
-                        base = ["location", "place", "venue"]
+                          base = ["location", "place", "venue"]
                         }])
   #_[canonical=["past team fixtures", "previous team matches", "past team games"]]
   #_[confirmation="past fixtures of $team_id"]
-  #_[result=["${our_team} ${result} ${opposition_team} at ${venue} for the ${league}. The score was ${our_score}-${opposition_score}"]]
+  #_[result=["${our_team} ${result} against ${opposition_team} at ${venue} for the ${league}. The score was ${our_score}-${opposition_score}"]]
   #[poll_interval=3600000ms]
   #[doc="Get the previous fixtures of the team"];
 
   monitorable list query topscorers(in req league_id: Entity(com.goal_smart:soccer_league) #_[prompt="what league do you want out check out"],
                         out player: String
                         #_[canonical = {
-                        base = ["person", "athlete", "teammate", "player"]
+                          base = ["person", "athlete", "teammate", "player"]
                         }],
                         out goals: Number
                         #_[canonical = {
-                        base = ["goals", "points", "score"]
+                          base = ["goals", "points", "score"]
                         }])
   #_[canonical=["top goal scorers", "goal scorers list", "league goal scorers"]]
   #_[confirmation="top scorers of $league_id"]
   #_[result=["${player} has scored ${goals} goals this season"]]
   #[poll_interval=3600000ms]
   #[doc="Get the top scorers of the league"];
 
+  monitorable list query topassisters(in req league_id: Entity(com.goal_smart:soccer_league) #_[prompt="what league do you want out check out"],
```

Comment:
I think you can consolidate this function, the previous one, and the next one with a single function that returns player statistics given a league ID. Then you can sort by goals, assists, red cards, yellow cards, etc.

The more you consolidate functions the less ambiguity you have in natural language, and the better accuracy you get.